retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs

```diff
@@ -0,0 +1,153 @@
+            DeleteObjectsFromBucket(bucketToDelete);
+            storageClient.DeleteBucket(bucketToDelete.Name);
+            Console.WriteLine($"Bucket {bucketToDelete.Name} is deleted.");
+        }
+        catch (Exception)
+        {
+            Console.WriteLine($"Bucket {bucketName} does not exist.");
+        }
+    }
+
+    /// <summary>Delete all objects from bucket.</summary>
+    private static void DeleteObjectsFromBucket(Bucket bucket)
+    {
+        Console.WriteLine($"Deleting object from bucket {bucket.Name}");
+
+        var blobs = storageClient.ListObjects(bucket.Name);
+        foreach (var blob in blobs)
+        {
+            storageClient.DeleteObject(bucket.Name, blob.Name);
+        }
+
+        Console.WriteLine($"All objects are deleted from a GCS bucket {bucket.Name}.");
+    }
+
+    /// <summary>Delete all products.</summary>
+    private static void DeleteAllProducts()
+    {
+        Console.WriteLine($"Deleting all products from catalog, please wait.");
+
+        var productServiceClient = ProductServiceClient.Create();
+        var listProductsRequest = new ListProductsRequest
+        {
+            Parent = defaultCatalog
+        };
+
+        var products = productServiceClient.ListProducts(listProductsRequest);
+        int deleteCount = 0;
+
+        foreach (var product in products)
+        {
+            var deleteProductRequest = new DeleteProductRequest
+            {
+                Name = product.Name
+            };
+
+            try
+            {
+                productServiceClient.DeleteProduct(deleteProductRequest);
+                deleteCount++;
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("Ignore PermissionDenied in case the product does not exist at time of deletion");
+            }
+        }
+
+        Console.WriteLine($"{deleteCount} products were deleted from {defaultCatalog}");
+    }
+
+    /// <summary>Delete Big Query dataset with tables.</summary>
+    private static void DeleteBQDatasetWithTables(string dataset)
+    {
+        string consoleOutput = string.Empty;
+        Console.WriteLine("Deleting a BigQuery dataset with all tables");
+
+        var deleteDatasetCommand = CurrentTerminalPrefix + CurrentTerminalQuotes + $"bq rm -r -d -f {dataset}" + CurrentTerminalQuotes;
+
+        var procStartInfo = new ProcessStartInfo(CurrentTerminalFile, deleteDatasetCommand)
+        {
+            RedirectStandardOutput = true,
+            UseShellExecute = false,
+            CreateNoWindow = true
+        };
+
+        using (Process process = new Process())
+        {
+            process.StartInfo = procStartInfo;
+            process.Start();
+
+            consoleOutput = process.StandardOutput.ReadToEnd();
+            Console.WriteLine(consoleOutput);
+        }
+    }
+
+    /// <summary>
+    /// Delete test resources.
+    /// </summary>
+    [TestResourcesSetupCleanup.Attributes.Example]
```

Comment:
Oh please don't reuse this attribute mechanism to run the setup. Is this somethig you expect tutorial users to have to run as well?

Reply:
no, this code is not for users, just for the person who will support the tests

Reply:
the attribute mechanism will be removed